### PR TITLE
[Reflection] Exclude CAS-related test from FieldInfoTests on Mono

### PR DESCRIPTION
**Pull request by MaximLipnin**
Relates to https://github.com/mono/mono/pull/10370

### CODE REVIEW

File: src/System.Runtime/tests/System/Reflection/FieldInfoTests.cs [hunk 1]
```diff
@@ -9,7 +9,11 @@
 
 namespace System.Reflection.Tests
 {
+#if MONO
```
Comment (marek-safar) [inline src/System.Runtime/tests/System/Reflection/FieldInfoTests.cs]: Why do we need to make it partial?

Comment (MaximLipnin) [reply src/System.Runtime/tests/System/Reflection/FieldInfoTests.cs]: To avoid a conflict with similar named class from System.Reflection

Comment (MaximLipnin) [reply src/System.Runtime/tests/System/Reflection/FieldInfoTests.cs]: ../../../external/corefx/src/System.Reflection/tests/FieldInfoTests.cs
../../../external/corefx/src/System.Runtime/tests/System/Reflection/FieldInfoTests.cs

Comment (marek-safar) [reply src/System.Runtime/tests/System/Reflection/FieldInfoTests.cs]: That seems to me as a problem on CoreFX side. All FieldInfoTests tests should be in System.Reflection. Please fix this upstream.

**Pull request merged**